### PR TITLE
[BugFix] Fix split constant non-ASCII source and empty delimiter

### PR DESCRIPTION
**Pull request by ZiheLiu**
The constant source and empty delimiter doesn't consider non-ASCII source.

```sql
MySQL> select split('膨胀', '');
'utf-8' codec can't decode byte 0xe8 in position 2: invalid continuation byte
```

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4

### CODE REVIEW

File: test/sql/test_function/T/test_split [hunk 1]
```diff
@@ -0,0 +1,10 @@
+-- name: test_split_constant_source_and_delimiter
+select split('测隔试隔试', '');
```
Comment (fzhedu) [inline test/sql/test_function/T/test_split:2]: should ut cover the tested bugs? as const input may not come out const columns

**Review by satanson — DISMISSED** [review]

File: test/sql/test_function/T/test_split (hunk 1 above)
Comment (ZiheLiu) [reply test/sql/test_function/T/test_split:2]: Fine.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 16 / 24 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exprs/split.cpp|3|11|27.27%|[43, 44, 45, 46, 47, 48, 50, 65]|
|:large_blue_circle:|src/util/utf8.h|13|13|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 16 / 24 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exprs/split.cpp|3|11|27.27%|[43, 44, 45, 46, 47, 48, 50, 65]|
|:large_blue_circle:|src/util/utf8.h|13|13|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 16 / 24 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exprs/split.cpp|3|11|27.27%|[43, 44, 45, 46, 47, 48, 50, 65]|
|:large_blue_circle:|src/util/utf8.h|13|13|100.00%|[]|

**Review by satanson — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**